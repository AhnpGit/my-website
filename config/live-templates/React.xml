<templateSet group="React">
  <template name="rx_module" value="const _ACTION_PREFIX = '$projectName$/$moduleName$'&#10;&#10;const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;&#10;// TODO: Connect the reducer.&#10;export default function reducer(state = {$defaultState$}, action) {&#10;    switch (action.type) {&#10;        case $ACTION_NAME$:&#10;            return {$END$}&#10;&#10;        default:&#10;            return state&#10;    }&#10;}&#10;&#10;export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$payload$},&#10;    meta: {},&#10;    error: undefined,&#10;})&#10;" description="Creates a DUCKS-module" toReformat="true" toShortenFQNames="true">
    <variable name="projectName" expression="" defaultValue="&quot;&lt;project name&gt;&quot;" alwaysStopAt="true" />
    <variable name="moduleName" expression="fileNameWithoutExtension()" defaultValue="" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <variable name="defaultState" expression="" defaultValue="" alwaysStopAt="true" />
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="payload" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component" value="import React from 'react'&#10;&#10;const $ComponentName$ = ({$props$}) =&gt; (&#10;    &lt;$END$/&gt;&#10;)&#10;&#10;export default $ComponentName$" description="Creates a functional component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="props" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component_class" value="import React from 'react'&#10;&#10;class $ComponentName$ extends React.Component {&#10;    render() {&#10;        return (&#10;            &lt;$END$/&gt;&#10;        )&#10;    }&#10;}&#10;&#10;export default $ComponentName$" description="Creates a class component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_proptypes" value="import PropTypes from 'prop-types'&#10;$ComponentName$.propTypes = {&#10;    $prop$: PropTypes.$END$,&#10;}" description="Adds proptypes to a component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="prop" expression="" defaultValue="&quot;prop&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action_async" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;const $ACTION_NAME$_PENDING = `${$ACTION_NAME$}_PENDING`&#10;const $ACTION_NAME$_FULFILLED = `${$ACTION_NAME$}_FULFILLED`&#10;const $ACTION_NAME$_REJECTED = `${$ACTION_NAME$}_REJECTED`" description="Creates async action types" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`" description="Creates an action type" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$END$},&#10;    meta: {},&#10;    error: undefined,&#10;})" description="Creates an action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator_async" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: getPayload(),&#10;    meta: {},&#10;    error: undefined,&#10;    &#10;    async function getPayload() {&#10;        return $END$&#10;    }&#10;})" description="Creates an async action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_container" value="import { connect } from 'react-redux'&#10;import $ComponentName$ from '../components/$ComponentName$'&#10;&#10;const mapStateToProps = (state) =&gt; ({&#10;    $END$&#10;})&#10;&#10;const mapDispatchToProps = (dispatch) =&gt; ({&#10;})&#10;&#10;export default connect(&#10;    mapStateToProps,&#10;    mapDispatchToProps,&#10;)($ComponentName$)" description="Creates a container class" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_styled" value="import styled from 'styled-components'&#10;&#10;const $ComponentName$ = styled.$ElementType$`&#10;  $END$&#10;`&#10;&#10;export default $ComponentName$" description="Creates a styled component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="ElementType" expression="" defaultValue="&quot;div&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
</templateSet>
