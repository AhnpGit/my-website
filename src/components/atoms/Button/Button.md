#### Button examples
    
    <div>
      <Button text="Default button" onClick={() => {}} />
      <br/>
      <Button type="button" appearance="primary" text="Primary button" onClick={() => {}} />
      <br/>
      <Button type="button" appearance="secondary" text="Secondary button" onClick={() => {}} />
      <br/>
      <Button type="button" appearance="success" text="Success button" onClick={() => {}} />
      <br/>
      <Button type="button" appearance="error" text="Error button" onClick={() => {}} />
      <br/>
      <Button type="button" appearance="primary" icon="search" onClick={() => {}} />
      <br/>
      <Button type="button" appearance="primary" icon="category" iconSize="xlarge" onClick={() => {}} />
      <br/>
      <Button type="button" appearance="primary" text="Button with Icon" icon="search" onClick={() => {}} />
      <br/>
      <Button type="button" appearance="secondary" text="Dropdown button with Icon" icon="search" hasDropdown={true} onClick={() => {}} />
    </div>
