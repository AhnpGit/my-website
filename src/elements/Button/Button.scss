@import '../../assets/sass/abstracts/functions';
@import '../../assets/sass/abstracts/variables';
@import '../../assets/sass/abstracts/mixins';

%baseButton {
  display: flex;
  appearance: none;
  position: relative;
  justify-content: center;
  align-items: center;
  text-align: center;
  vertical-align: middle;
  padding: map-deep-get($button, 'padding', 'vertical') map-deep-get($button, 'padding', 'horizontal');
  margin: 0 6px;
  color: currentColor;
  border: 0;
  border-radius: map_get($button, 'borderRadius');
  font-family: 'Lato', helvetica, arial, sans-serif;
  font-size: 80%;
  font-weight: map_get($button, 'fontWeight');
  line-height: map-deep-get($base, 'icon', 'small');
  text-decoration: none;
  text-overflow: ellipsis;
  overflow: hidden;
  cursor: pointer;
  transition: background-color 0.2s, border 0.2s, color 0.2s;

  svg + span, span + svg, svg + svg {
    margin-left: map_get($base, 'spacer');
  }
}

%primary {
  color: map_get($colors, 'pureWhite');
  border: 1px solid map-deep-get($button, 'borderColor', 'primary');
  background-color: map-deep-get($button, 'backgroundColor', 'primary');
  box-shadow: inset 0 -1px 0 0 map-deep-get($button, 'borderColor', 'primary');

  @include on-event {
    background-color: map-deep-get($button, 'hoverBackgroundColor', 'primary');
    text-decoration: none;
    box-shadow: inset 0 2px 3px 0 rgba(0,0,0,0.13),
                inset 0 -1px 0 0 map-deep-get($button, 'borderColor', 'primary');
  }
}

%secondary {
  color: map_get($font, 'color');
  background-color: map-deep-get($button, 'backgroundColor', 'secondary');
  box-shadow: inset 0 -1px 0 0 map-deep-get($button, 'borderColor', 'secondary');

  @include on-event {
    color: map_get($colors, 'mineShaft');
    background-color: map-deep-get($button, 'hoverBackgroundColor', 'secondary');
    box-shadow: inset 0 2px 3px 0 rgba(0,0,0,0.13),
                inset 0 -1px 0 0 map-deep-get($button, 'borderColor', 'secondary');
  }
}

%success {
  color: map_get($colors, 'aquaHaze');
  text-shadow: none;
  background-color: map-deep-get($button, 'backgroundColor', 'success');
  box-shadow: inset 0 -1px 0 0 map-deep-get($button, 'borderColor', 'success');

  &:hover {
    box-shadow: inset 0 2px 3px 0 rgba(0,0,0,0.13),
                inset 0 -1px 0 0 map-deep-get($button, 'borderColor', 'success');
  }
}

%error {
  color: map_get($colors, 'aquaHaze');
  text-shadow: none;
  background-color: map-deep-get($button, 'backgroundColor', 'error');
  box-shadow: inset 0 -1px 0 0 map-deep-get($button, 'borderColor', 'error');

  &:hover {
    box-shadow: inset 0 2px 3px 0 rgba(0,0,0,0.13),
                inset 0 -1px 0 0 map-deep-get($button, 'borderColor', 'error');
  }
}

%close {
  background: transparent;
  color: map_get($colors, 'black');
  box-shadow: none;

  &:hover {
    background: transparent;
    box-shadow: none;
  }
}

.btn {
  &--primary {
    @extend %baseButton;
    @extend %primary;
  }
  &--secondary {
    @extend %baseButton;
    @extend %secondary;
  }
  &--success {
    @extend %baseButton;
    @extend %success;
  }
  &--error {
    @extend %baseButton;
    @extend %error;
  }
  &--close {
    @extend %baseButton;
    @extend %close;
  }
}

.btn__text {
  display: inline-flex;
  align-items: center;
  align-content: center;
}
