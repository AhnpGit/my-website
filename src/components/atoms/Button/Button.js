// @flow

import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import TextLabel from '../../atoms/TextLabel/TextLabel'
import { renderButton } from '../../../utils/style/mixins'
import { base } from '../../../utils/style/branding'
import Icon from '../Icon/Icon'

type ButtonProps = {
  appearance?: 'primary' | 'secondary' | 'success' | 'error' | 'close',
  type?: 'button' | 'submit' | 'reset',
  text?: string | React.Element<any>,
  icon?: string,
  iconSize?: 'xsmall' | 'small' | 'normal' | 'large' | 'xlarge',
  hasDropdown?: boolean,
  onClick: () => mixed,
}
/**
 * Buttons are generated by the renderButton mixin. This will render an object with button styling (so you can also
 * use this for a Link that needs to look like a button, but still stay semantically correct). Currently we have 5
 * button appearance: primary(default), secondary, success, error and close (to be used for modals/popups).
 */
const Button = ({ type = 'button',
                  appearance = 'primary',
                  hasDropdown = false,
                  icon,
                  iconSize = 'small',
                  text,
                  onClick}: ButtonProps) => (
  <Component type={ type } appearance={ appearance } className={ className } onClick={ onClick }>
    { icon ? <Icon name={ icon } fixedSize={ iconSize }/> : null }
    { text ? <TextLabel text={ text } /> : null }
    { hasDropdown ? <Icon name="triangle-down" fixedSize="xsmall" /> : null }
  </Component>
)

const Component = styled.button`
  ${ props => renderButton[props.appearance] || renderButton.primary };
  
  svg + span, span + svg, svg + svg {
    margin-left: ${ base.spacer };
  }
`

Button.propTypes = {
  /**
   * The classname is required by Styled Components to allow overrides of the component
   */
  className: PropTypes.string,
  /**
   * The appearance of the button. Available appearances: primary(default), secondary, success, error and close
   */
  appearance: PropTypes.oneOf(['primary', 'secondary', 'success', 'error', 'close']),
  /**
   * The type of teh button. Available options: button, submit and reset.
   */
  type: PropTypes.oneOf(['button', 'submit', 'reset']),
  /**
   * The button text
   */
  text: PropTypes.string,
  /**
   * This will render an icon on the button (not for dropdown arrows)
   */
  icon: PropTypes.string,
  /**
   * Set the size of the icon. Options: xsmall(8px), small(16px), normal(32px), large(48px), xlarge(64px). Size are
   * controlled in the branding util.
   */
  iconSize: PropTypes.oneOf(['xsmall', 'small', 'normal', 'large', 'xlarge']),
  /**
   * Boolean value that will render a dropdown triangle when set to true.
   */
  hasDropdown: PropTypes.bool,
  /**
   * The onClick parameter makes it possible to add events to the component
   */
  onClick: PropTypes.func.isRequired,
}

export default Button
