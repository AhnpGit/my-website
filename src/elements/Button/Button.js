// @flow

import React from 'react'
import PropTypes from 'prop-types'

import Icon from '../Icon/Icon'

type ButtonProps = {
  appearance?: 'primary' | 'secondary' | 'success' | 'error' | 'close',
  hasDropdown?: boolean,
  icon?: string,
  iconSize?: 'xsmall' | 'small' | 'normal' | 'large' | 'xlarge',
  onClick: () => mixed,
  text?: string | React.Element<any>,
  type?: 'button' | 'submit' | 'reset',
}

/**
 * Buttons are generated by the renderButton mixin. This will render an object with button styling (so you can also
 * use this for a Link that needs to look like a button, but still stay semantically correct). Currently we have 5
 * button appearance: primary(default), secondary, success, error and close (to be used for modals/popups).
 */
const Button = ({
                  appearance = 'primary',
                  hasDropdown = false,
                  icon,
                  iconSize = 'small',
                  onClick,
                  text,
                  type = 'button',
                }: ButtonProps) => (
  <button type={type} className={`btn--${appearance}`} onClick={onClick}>
    {icon && <Icon name={icon} fixedSize={iconSize} />}
    {text && <span className="btn__text">{text}</span>}
    {hasDropdown && <Icon name="triangle-down" fixedSize="xsmall" />}
  </button>
)

Button.propTypes = {
  /**
   * The appearance of the button. Available appearances: primary(default), secondary, success, error and close
   */
  appearance: PropTypes.oneOf([ 'primary', 'secondary', 'success', 'error', 'close' ]),

  /**
   * Boolean value that will render a dropdown triangle when set to true.
   */
  hasDropdown: PropTypes.bool,

  /**
   * This will render an icon on the button (not for dropdown arrows)
   */
  icon: PropTypes.string,

  /**
   * Set the size of the icon. Options: xsmall(8px), small(16px), normal(32px), large(48px), xlarge(64px). Size are
   * controlled in the branding util.
   */
  iconSize: PropTypes.oneOf([ 'xsmall', 'small', 'normal', 'large', 'xlarge' ]),

  /**
   * The onClick parameter makes it possible to add events to the component
   */
  onClick({ onClick, type }: ButtonProps, propName: string, componentName: string) {
    if (type !== 'submit' && onClick === undefined) {
      return new Error(`Failed prop type: The prop '${propName}' is marked as required in '${componentName}' when the prop 'type' is not 'submit', but its value is 'undefined'.`)
    }
  },

  /**
   * The type of teh button. Available options: button, submit and reset.
   */
  type: PropTypes.oneOf([ 'button', 'submit', 'reset' ]),

  /**
   * The button text
   */
  text: PropTypes.string,
}

export default Button
