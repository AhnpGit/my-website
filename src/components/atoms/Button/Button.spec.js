import React from 'react'
import renderer from 'react-test-renderer'
import Button from './Button'

let nop = () => {}

describe('The Button component', () => {

  it(`Renders a primary button when no arguments are passed`, () => {
    const button = renderer.create(
      <Button onClick={nop} />
    ).toJSON()
    expect(button).toMatchSnapshot()
  })

  it(`Renders a primary button when type is primary`, () => {
    const button = renderer.create(
      <Button appearance="primary" text="Button" onClick={nop} />
    ).toJSON()
    expect(button).toMatchSnapshot()
  })

  it(`Renders a secondary button when type is secondary`, () => {
    const button = renderer.create(
      <Button appearance="secondary" text="Button" onClick={nop} />
    ).toJSON()
    expect(button).toMatchSnapshot()
  })

  it(`Renders a success button when type is success`, () => {
    const button = renderer.create(
      <Button appearance="success" text="Button" onClick={nop} />
    ).toJSON()
    expect(button).toMatchSnapshot()
  })

  it(`Renders a error button when type is error`, () => {
    const button = renderer.create(
      <Button appearance="error" text="Button" onClick={nop} />
    ).toJSON()
    expect(button).toMatchSnapshot()
  })

  it(`Renders a button which has just an icon`, () => {
    const button = renderer.create(
      <Button icon="search" onClick={nop} />
    ).toJSON()
    expect(button).toMatchSnapshot()
  })

  it(`Renders a button which has just a xsmall icon`, () => {
    const button = renderer.create(
      <Button icon="search" iconSize="xsmall" onClick={nop} />
    ).toJSON()
    expect(button).toMatchSnapshot()
  })

  it(`Renders a button which has just a small icon`, () => {
    const button = renderer.create(
      <Button icon="search" iconSize="small" onClick={nop} />
    ).toJSON()
    expect(button).toMatchSnapshot()
  })

  it(`Renders a button which has just a normal icon`, () => {
    const button = renderer.create(
      <Button icon="search" iconSize="normal" onClick={nop} />
    ).toJSON()
    expect(button).toMatchSnapshot()
  })

  it(`Renders a button which has just a large icon`, () => {
    const button = renderer.create(
      <Button icon="search" iconSize="large" onClick={nop} />
    ).toJSON()
    expect(button).toMatchSnapshot()
  })

  it(`Renders a button which has just a xlarge icon`, () => {
    const button = renderer.create(
      <Button icon="search" iconSize="xlarge" onClick={nop} />
    ).toJSON()
    expect(button).toMatchSnapshot()
  })

  it(`Renders a button which has dropdown arrow`, () => {
    const button = renderer.create(
      <Button onClick={nop} hasDropdown={true} />
    ).toJSON()
    expect(button).toMatchSnapshot()
  })

  it(`Renders a button which has an icon, text and dropdown arrow`, () => {
    const button = renderer.create(
      <Button icon="search" text="Button" hasDropdown={true} onClick={nop} />
    ).toJSON()
    expect(button).toMatchSnapshot()
  })

})
