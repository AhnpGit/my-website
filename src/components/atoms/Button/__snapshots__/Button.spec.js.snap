// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Button component Renders a button which has an icon, text and dropdown arrow 1`] = `
<button
  className="sc-htpNat ivUSgc"
  onClick={[Function]}
  type="button"
>
  <svg
    aria-labelledby="icon-search-title"
    className="sc-bwzfXH eyjyQn"
    role="img"
  >
    <title>
      search
    </title>
    <use
      pointerEvents="none"
      xlinkHref="#icons-search"
    />
  </svg>
  <span
    className="sc-bdVaJa hJMgGI"
  >
    <span
      className="Text"
    >
      Button
    </span>
  </span>
  <svg
    aria-labelledby="icon-triangle-down-title"
    className="sc-bwzfXH hqyKWH"
    role="img"
  >
    <title>
      triangle-down
    </title>
    <use
      pointerEvents="none"
      xlinkHref="#icons-triangle-down"
    />
  </svg>
</button>
`;

exports[`The Button component Renders a button which has dropdown arrow 1`] = `
<button
  className="sc-htpNat ivUSgc"
  onClick={[Function]}
  type="button"
>
  <svg
    aria-labelledby="icon-triangle-down-title"
    className="sc-bwzfXH hqyKWH"
    role="img"
  >
    <title>
      triangle-down
    </title>
    <use
      pointerEvents="none"
      xlinkHref="#icons-triangle-down"
    />
  </svg>
</button>
`;

exports[`The Button component Renders a button which has just a large icon 1`] = `
<button
  className="sc-htpNat ivUSgc"
  onClick={[Function]}
  type="button"
>
  <svg
    aria-labelledby="icon-search-title"
    className="sc-bwzfXH bKdoCw"
    role="img"
  >
    <title>
      search
    </title>
    <use
      pointerEvents="none"
      xlinkHref="#icons-search"
    />
  </svg>
</button>
`;

exports[`The Button component Renders a button which has just a normal icon 1`] = `
<button
  className="sc-htpNat ivUSgc"
  onClick={[Function]}
  type="button"
>
  <svg
    aria-labelledby="icon-search-title"
    className="sc-bwzfXH fOPIKn"
    role="img"
  >
    <title>
      search
    </title>
    <use
      pointerEvents="none"
      xlinkHref="#icons-search"
    />
  </svg>
</button>
`;

exports[`The Button component Renders a button which has just a small icon 1`] = `
<button
  className="sc-htpNat ivUSgc"
  onClick={[Function]}
  type="button"
>
  <svg
    aria-labelledby="icon-search-title"
    className="sc-bwzfXH eyjyQn"
    role="img"
  >
    <title>
      search
    </title>
    <use
      pointerEvents="none"
      xlinkHref="#icons-search"
    />
  </svg>
</button>
`;

exports[`The Button component Renders a button which has just a xlarge icon 1`] = `
<button
  className="sc-htpNat ivUSgc"
  onClick={[Function]}
  type="button"
>
  <svg
    aria-labelledby="icon-search-title"
    className="sc-bwzfXH bpaUhd"
    role="img"
  >
    <title>
      search
    </title>
    <use
      pointerEvents="none"
      xlinkHref="#icons-search"
    />
  </svg>
</button>
`;

exports[`The Button component Renders a button which has just a xsmall icon 1`] = `
<button
  className="sc-htpNat ivUSgc"
  onClick={[Function]}
  type="button"
>
  <svg
    aria-labelledby="icon-search-title"
    className="sc-bwzfXH hqyKWH"
    role="img"
  >
    <title>
      search
    </title>
    <use
      pointerEvents="none"
      xlinkHref="#icons-search"
    />
  </svg>
</button>
`;

exports[`The Button component Renders a button which has just an icon 1`] = `
<button
  className="sc-htpNat ivUSgc"
  onClick={[Function]}
  type="button"
>
  <svg
    aria-labelledby="icon-search-title"
    className="sc-bwzfXH eyjyQn"
    role="img"
  >
    <title>
      search
    </title>
    <use
      pointerEvents="none"
      xlinkHref="#icons-search"
    />
  </svg>
</button>
`;

exports[`The Button component Renders a error button when type is error 1`] = `
<button
  className="sc-htpNat eKoXz"
  onClick={[Function]}
  type="button"
>
  <span
    className="sc-bdVaJa hJMgGI"
  >
    <span
      className="Text"
    >
      Button
    </span>
  </span>
</button>
`;

exports[`The Button component Renders a primary button when no arguments are passed 1`] = `
<button
  className="sc-htpNat ivUSgc"
  onClick={[Function]}
  type="button"
/>
`;

exports[`The Button component Renders a primary button when type is primary 1`] = `
<button
  className="sc-htpNat ivUSgc"
  onClick={[Function]}
  type="button"
>
  <span
    className="sc-bdVaJa hJMgGI"
  >
    <span
      className="Text"
    >
      Button
    </span>
  </span>
</button>
`;

exports[`The Button component Renders a secondary button when type is secondary 1`] = `
<button
  className="sc-htpNat fpOstT"
  onClick={[Function]}
  type="button"
>
  <span
    className="sc-bdVaJa hJMgGI"
  >
    <span
      className="Text"
    >
      Button
    </span>
  </span>
</button>
`;

exports[`The Button component Renders a success button when type is success 1`] = `
<button
  className="sc-htpNat gGHtYq"
  onClick={[Function]}
  type="button"
>
  <span
    className="sc-bdVaJa hJMgGI"
  >
    <span
      className="Text"
    >
      Button
    </span>
  </span>
</button>
`;
